node/src/filecheckpointer.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import fs from 'fs';
+import { Checkpointer } from './checkpointer';
+
+export interface CheckpointerState {
```

Comment:
I think this should be used only within the file checkpointer as its persistent data format. It probably shouldn't be exported from this module